src/Migrate.php

```diff
@@ -0,0 +1,795 @@
+                        )),
+                        'assignedFutureGrants' => $this->sourceConnection->fetchAll(sprintf(
+                            'SHOW FUTURE GRANTS TO ROLE %s;',
+                            $role['name']
+                        )),
+                    ]
+                );
+            }
+        }
+        return $tmp;
+    }
+
+    public function createMainRole(array $mainRoleWithGrants, array $databases, array $users): void
+    {
+        $user = $mainRole = $mainRoleWithGrants['name'];
+
+        $this->destinationConnection->createRole(['name' => $mainRole, 'privilege' => 'OWNERSHIP']);
+
+        $mainRoleGrants = [
+            'GRANT CREATE DATABASE ON ACCOUNT TO ROLE %s;',
+            'GRANT CREATE ROLE ON ACCOUNT TO ROLE %s WITH GRANT OPTION;',
+            'GRANT CREATE USER ON ACCOUNT TO ROLE %s WITH GRANT OPTION;',
+        ];
+
+        foreach ($mainRoleGrants as $mainRoleGrant) {
+            $this->destinationConnection->query(sprintf($mainRoleGrant, $mainRole));
+        }
+
+        $warehouses = array_filter($mainRoleWithGrants['assignedGrants'], fn($v) => $v['granted_on'] === 'WAREHOUSE');
+        $useWarehouse = false;
+        foreach ($warehouses as $warehouse) {
+            $warehouseSize = self::createWarehouse($warehouse);
+            $this->destinationConnection->assignGrantToRole($warehouse);
+
+            if ($useWarehouse === false || $warehouseSize === 'X-Small') {
+                $useWarehouse = $warehouse;
+            }
+        }
+
+        $this->destinationConnection->query(sprintf(
+            'USE WAREHOUSE %s',
+            QueryBuilder::quoteIdentifier($useWarehouse['name'])
+        ));
+
+        $this->destinationConnection->query(sprintf(
+            'CREATE USER %s PASSWORD=%s DEFAULT_ROLE=%s',
+            $user,
+            QueryBuilder::quote($users[$user]),
+            $mainRole
+        ));
+
+        $this->destinationConnection->query(sprintf(
+            'GRANT ROLE %s TO USER %s;',
+            $mainRole,
+            $user
+        ));
+
+        $projectUsers = array_filter($mainRoleWithGrants['assignedGrants'], function ($v) use ($databases) {
+            if ($v['privilege'] !== 'OWNERSHIP') {
+                return false;
+            }
+            if ($v['granted_on'] !== 'USER') {
+                return false;
+            }
+            return in_array($v['name'], $databases);
+        });
+
+        foreach ($projectUsers as $projectUser) {
+            self::createUser($projectUser, $users);
+            $this->destinationConnection->assignGrantToRole($projectUser);
+        }
+
+        $this->destinationConnection->useRole('ACCOUNTADMIN');
+
+        $this->destinationConnection->query(sprintf(
+            'GRANT ROLE %s TO ROLE SYSADMIN;',
+            $mainRole
+        ));
+    }
+
+    public function cleanupProject(): void
+    {
+        $this->destinationConnection->useRole('ACCOUNTADMIN');
```

Comment:
Jen pro jistotu, aby to tady nezapoměl - možná nemáš accountadmin roli - možná máš jinou co ti udělal klient.

Reply:
pošlu název role přes konfiguraci

Reply:
hotovo - https://github.com/keboola/project-migration-tool/pull/1/commits/9fb3ba195b79225c48e05855ccf4e5563d92aa08